### PR TITLE
Add full UIDevice support [WIP]

### PR DESCRIPTION
**Pull request by GertjanReynaert**
DO NOT MERGE! Still work in progress!
- [x] add all possible UIDevice values to exported constants
- [x] mark dynamic properties with `initial` prefix so that it is clear these values don't update
- [x] export methods for orientationState, batteryState, batteryLevel and proximityState to manually request new value
- [x] implement watch method, so that callback gets called on every value change/native notification
- [x] add stopWatching methods, so you can remove the eventlistener

### CODE REVIEW

**Comment by GertjanReynaert** [comment]
@kkjdaniel Would you mind reviewing this pr?

**Comment by kkjdaniel** [comment]
Lots of changes there! Yeah sure I'll take a look, I'm guessing ideally it should export the events so that the values can be updated in the app and not just at the start of the runtime?

**Comment by GertjanReynaert** [comment]
Still work in progress!

- I've marked the dynamic properties that are exported as initialValues so that it is clear these values don't update
- I've exported a few methods so you can manually re-request dynamic properties (orientation/batteryState/batteryLevel/proximityState)
- I've started implementing watch functions which should pass updated values to the callback on every native notification (WIP)
- After the watch methods are fully functional, I would like to create some kind of stopWatching method so you can disable the watch methods too.

**Comment by kkjdaniel** [comment]
Been really busy recently, will hopefully get a chance to review this PR and little more today - and I'll look into those remaining features, unless you've already made progress on them?

File: Device.js [hunk 1]
```diff
@@ -9,14 +10,87 @@ class Device {
...
+
+  getBatteryLevel(callback) {
+    DeviceUtil.getBatteryLevel(callback);
+  }
+
+  getProximityState(callback) {
+    DeviceUtil.getProximityState(callback);
+  }
+
+  watchOrientationChange(callback) {
```
Comment (imevro) [inline Device.js:52]: Hm, maybe better use `on*Event*` instead of `watch*Event*`?

**Comment by aliemre** [comment]
@kkjdaniel @GertjanReynaert Is there any update for the PR? Other features of UIDevice is important. I'd be appreciate when it's merged.

**Pull request closed without merging**